nomad/state/state_store.go

```diff
@@ -5366,6 +5366,45 @@ func (s *StateStore) DeleteOneTimeTokens(msgType structs.MessageType, index uint
 	txn := s.db.WriteTxnMsgT(msgType, index)
 	defer txn.Abort()
 
+	err := s.deleteOneTimeTokensTxn(txn, index, ids)
+	if err != nil {
+		return err
+	}
+	return txn.Commit()
+}
+
+// ExpireOneTimeTokens deletes tokens that have expired
+func (s *StateStore) ExpireOneTimeTokens(msgType structs.MessageType, index uint64) error {
+	txn := s.db.WriteTxnMsgT(msgType, index)
+	defer txn.Abort()
+
+	iter, err := s.oneTimeTokensExpiredTxn(txn, nil)
+	if err != nil {
+		return err
+	}
+
+	ids := []string{}
+	for {
+		raw := iter.Next()
+		if raw == nil {
+			break
+		}
+		ott, ok := raw.(*structs.OneTimeToken)
+		if !ok || ott == nil {
+			return fmt.Errorf("could not decode one-time token")
+		}
+		ids = append(ids, ott.AccessorID)
+	}
+
+	err = s.deleteOneTimeTokensTxn(txn, index, ids)
```

Comment:
The invariants should hold (see my other comment on the schema-level enforcement). So rather than having a separate delete function we could really just replace the delete function that takes IDs and swap it out for OTT and it'd still be correct but also more efficient. Will fix.

Reply:
Done in 59151ea. Because of the 1:1 invariant this mostly just saves looping twice over the list of expired tokens (once to get the IDs and once to delete them), but that's worth a tiny bit of code repetition.